calblueprint/meepanyar
# Customer Overview Sorting and Filtering

[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR

Closes issue described [here](https://www.notion.so/Customer-Overview-sort-feature-f7567cc56efe4785a6dc470fb9d63282). In summary, we add support for sorting and filtering results in the customer overview page. 

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

## How to review

1. [Setup repository locally](https://github.com/calblueprint/meepanyar/blob/main/README.md). 
2. Run `npm start`. Navigate to Customer Tab (2nd to the left). 
3. Select your sort by and filter by attributes. 

[//]: # 'The order in which to review files and what to expect when testing locally'

### Online sources
- [Multiselect from MaterialUI](https://material-ui.com/components/selects/#multiple-select)

[//]: # 'Optional - copy links to any tutorial or documentation that was useful to you when working on this PR'

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"

### Screenshots
Users can now navigate to the customer overview screen and see their results sorted alphabetically and grouped by active status by default. In the following example, results are grouped by meter status.  
![Screen Shot 2021-02-23 at 5 16 39 PM](https://user-images.githubusercontent.com/44735047/108930547-f8b7c900-75fa-11eb-88d5-139e5868c365.png)


Users can specify how to sort and filter the customer results by using this multi-select dropdown. 
![Screen Shot 2021-02-23 at 11 48 03 AM](https://user-images.githubusercontent.com/44735047/108899693-8ed4fa80-75cd-11eb-83da-dfbb73881e40.png)


[//]: # "Add screenshots of expected behavior - GIFs if you're feeling fancy!"

CC: @julianrkung
